Lightyear/StringFile.idr

```diff
@@ -0,0 +1,36 @@
+-- -------------------------------------------------------------- [ StringFile ]
+-- Module      : Lightyear.StringFile
+-- Description : String-related effectful file parsers.
+--
+-- This code is distributed under the BSD 2-clause license.
+-- See the file LICENSE in the root directory for its full text.
+-- --------------------------------------------------------------------- [ EOH ]
+
+module Lightyear.StringFile
+
+import public Effects
+import public Effect.File
+
+import Lightyear
+import Lightyear.Strings
+
+-- ---------------------------------------------------- [ A String File Parser ]
+
+parseFile : Parser a
+         -> String
+         -> Eff (Either String a) [FILE_IO ()]
+parseFile p fn =
+    case !(open fn Read) of
+      True => do
+        src <- doRead ""
+        close
+        pure $ parse p src
+      False => pure $ Left "File Not Found"
```

Comment:
actually, maybe it might be best to introduce an error data type to aide better in fault diagnosis rather than looking at the strings.